header-only/layout-builder/awkward/LayoutBuilder.h

```diff
@@ -1151,6 +1151,14 @@ namespace awkward {
         return content_;
       }
 
+      /// @brief Inserts an explicit value in the `index` buffer and
+      /// returns the reference to the builder content.
+      BUILDER&
+      append_index(size_t i) noexcept {
+        index_.append(i);
+        return content_;
+      }
```

Comment:
I think this is a necessary method for categorical

Reply:
Okay!